crates/router/src/core/payments/transformers.rs

```diff
@@ -876,6 +884,18 @@ pub fn qr_code_next_steps_check(
     Ok(qr_code_instructions)
 }
 
+pub fn fetch_qr_code_url_next_steps_check(
+    payment_attempt: storage::PaymentAttempt,
+) -> RouterResult<Option<api_models::payments::FetchQrCodeInformation>> {
+    let qr_code_steps: Option<Result<api_models::payments::FetchQrCodeInformation, _>> =
+        payment_attempt
+            .connector_metadata
+            .map(|metadata| metadata.parse_value("FetchQrCodeInformation"));
+
+    let qr_code_fetch_url = qr_code_steps.transpose().ok().flatten();
```

Comment:
why are we ignoring the error in case of parse failures?

Reply:
This function is used to decide the next action,  if `fetch_qr_code_url_next_steps_check` parses connector_metadata to FetchQrCodeInformation. We mark next action into `fetch_qr_code_information`. If it doesn't parse, that implies next action is not `fetch_qr_code_information`